### PR TITLE
Sound Management

### PR DESCRIPTION
**Pull request by cothon01**
Welcome... to the _future_ of AUDIO MANAGEMENT!

I created a sound management system that facilitates the playing of sound clips (One shot, not looping). Why this is useful:
-> Removes inspector bloat: Your gameobjects don't need to store their own AudioSource components to play sounds
-> Easy Access: Uses a singleton (SFXManager) that can be used by any script with a single line of code
-> Uninterrupted Sound Playing: AudioSources are like a resource. You need more than one to play multiple sounds simultaneously. SFXManager, manages a bunch of them and ensures that no sound is interrupted (unless you play a ridiculous number of simultaneous sounds)
-> Supports 3D sound: SFXManager moves the AudioSource to a target point in case you need 3d sound.

I also included an AudioMixer with two sound channels (BGM and SFX). And a test scene that you can test this system with.

Bonus: I added SceneOnlyEdit.cs which is a modified version of the ReadOnly property drawer. By using the [SceneOnlyEdit] attribute, you mark an inspector variable so that it cannot be edited during Play Mode. But you can edit its value when the game is not running.

### CODE REVIEW

**Comment by Helcostr** [comment]
oooo fancy.

**Review by Sneakydeaky — APPROVED** [review]
Wow, That's Awesome! Excellent job.

**Comment by Helcostr** [comment]
There seems to be an issue when the source amount is too low, currently playing sounds get overwritten. Is this intentional? Are you going to dynamically allocate more AudioSources?

File: Assets/Scripts/Audio/SFXManager.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        AudioSource source = sources[0];
+        for (int i = 0; i < sources.Length; i++)
+        {
+            if (sources[i].isPlaying == false)
+            {
+                source = sources[i];
+                break;
+            }
+        }
+        return source;
```
Comment (Helcostr) [inline Assets/Scripts/Audio/SFXManager.cs]: This will return sources[0] if all sources are preoccupied.

**Review by Helcostr — CHANGES_REQUESTED** [review]
Please check and approve/change the commented line.

File: Assets/Scripts/Audio/SFXManager.cs (hunk 1 above)
Comment (cothon01) [reply Assets/Scripts/Audio/SFXManager.cs]: Just a minor setback ^-^' ...

**Review by Helcostr — APPROVED** [review]
No issues.

**Pull request merged**